modules/renter/contractor/formcontract.go

```diff
@@ -363,6 +367,7 @@ func (c *Contractor) formContracts(a modules.Allowance) error {
 			// negotiate with because they'll probably be more expensive than
 			// we can afford.
 			c.log.Println("WARN: failed to negotiate contract:", h.NetAddress, err)
+			continue
 		}
 		if numContracts++; numContracts >= a.Hosts {
 			break
```

Comment:
Don't we have insufficient hosts if `numContracts < a.Hosts` when the for loop exits? Should we return an error in that case?